terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -94,25 +94,105 @@ def update_registry_declaration(new_req, old_req, updated_content)
         end
       end
 
-      def update_lockfile_declaration(updated_manifest_files) # rubocop:disable Metrics/AbcSize
+      def extract_provider_h1_hashes(content, declaration_regex)
+        content.match(declaration_regex).to_s.
+          match(hashes_object_regex).to_s.
+          split("\n").map { |hash| hash.match(hashes_string_regex).to_s }.
+          select { |h| h&.match?(/^h1:/) }
+      end
```

Comment:
Or a simpler version that doesn't check the preceding `hashes` key:

```suggestion
      H1_HASH_REGEX = /\"h1:([\w\d\=\/]+)\"/.freeze
      
      def extract_provider_h1_hashes(content, declaration_regex)
        content.match(declaration_regex)
               .string
               .scan(H1_HASH_REGEX)
               .flatten
      end
```

Reply:
Giving this a bit more thought, it might be nice to create a proper abstraction for Terraform lockfiles. Moving all of the business logic around extracting information and into methods in a new class would help keep `FileUpdater` small and focused, and make it easier to test this functionality.

Reply:
I think that would be ideal. Perhaps in a follow-up PR though?

Reply:
You have a working solution, which is a good first step. But the current implementation needs to disable `Metrics/AbcSize` to pass linting checks in a few places, so I think this PR would benefit from a quick refactoring pass before merging. I'd be happy to pair with you on that — just let me know! 😄

Reply:
This is a refactoring pass on some code already merged into Core, so you're right that we should focus on "making it right" and address linting checks this pass.

I'll ping you to set up some time to pair!